catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/ShareableMetacardPolicyExtension.java

```diff
@@ -61,39 +61,48 @@ public ShareableMetacardPolicyExtension(
         .collect(Collectors.toList());
   }
 
-  private Map<String, Set<String>> groupPermissionsByKey(List<KeyValuePermission> permissions) {
+  private static Map<String, Set<String>> groupPermissionsByKey(
+      List<KeyValuePermission> permissions) {
     return permissions
         .stream()
         .collect(
             Collectors.toMap(
                 KeyValuePermission::getKey, KeyValuePermission::getValues, Sets::union));
   }
 
-  private Predicate<CollectionPermission> system() {
+  private boolean system(CollectionPermission subject) {
     Set<String> values = ImmutableSet.of(config.getSystemUserAttributeValue());
     KeyValuePermission perm = new KeyValuePermission(config.getSystemUserAttribute(), values);
-    return (subject) -> subject.implies(perm);
+    return subject.implies(perm);
   }
 
-  private Predicate<CollectionPermission> predicate(
-      Map<String, Set<String>> permissions, String k1, String k2) {
-    if (!permissions.containsKey(k1)) {
-      return (subject) -> false;
+  private static boolean evaluate(
+      CollectionPermission subject,
+      Map<String, Set<String>> permissions,
+      String permissionKey,
+      String attributeName) {
+    if (!permissions.containsKey(permissionKey)) {
+      return false;
     }
-    return (subject) -> subject.implies(new KeyValuePermission(k2, permissions.get(k1)));
+    return subject.implies(new KeyValuePermission(attributeName, permissions.get(permissionKey)));
   }
 
-  private Predicate<CollectionPermission> owner(Map<String, Set<String>> permissions) {
-    return predicate(permissions, Core.METACARD_OWNER, subjectIdentity.getIdentityAttribute());
+  private boolean owner(CollectionPermission subject, Map<String, Set<String>> permissions) {
+    return evaluate(
+        subject, permissions, Core.METACARD_OWNER, subjectIdentity.getIdentityAttribute());
   }
 
-  private Predicate<CollectionPermission> individuals(Map<String, Set<String>> permissions) {
-    return predicate(
-        permissions, SecurityAttributes.ACCESS_INDIVIDUALS, subjectIdentity.getIdentityAttribute());
+  private boolean individuals(CollectionPermission subject, Map<String, Set<String>> permissions) {
+    return evaluate(
+        subject,
+        permissions,
+        SecurityAttributes.ACCESS_INDIVIDUALS,
+        subjectIdentity.getIdentityAttribute());
   }
 
-  private Predicate<CollectionPermission> groups(Map<String, Set<String>> permissions) {
-    return predicate(permissions, SecurityAttributes.ACCESS_GROUPS, Constants.ROLES_CLAIM_URI);
+  private boolean groups(CollectionPermission subject, Map<String, Set<String>> permissions) {
+    return evaluate(
+        subject, permissions, SecurityAttributes.ACCESS_GROUPS, Constants.ROLES_CLAIM_URI);
   }
 
   private KeyValueCollectionPermission isPermitted(
```

Comment:
❗️ This method needs to know if the permission collection is a match one or all collection.